client/sections.js

```diff
@@ -14,29 +14,27 @@ const extensions = require( 'extensions' );
 const extensionSections = extensions.map( extension => {
 	try {
 		const pkg = JSON.parse( fs.readFileSync( path.join( __dirname, 'extensions', extension, 'package.json' ) ) );
-		return pkg.env_id.indexOf( config( 'env_id' ) ) > -1
-			? pkg.section
-			: null;
+
+		return Object.assign( {}, pkg.section, { envId: pkg.env_id } );
 	} catch ( e ) {
 		return null;
 	}
 } );
 
-if ( config.isEnabled( 'devdocs' ) ) {
-	sections.push( {
-		name: 'devdocs',
-		paths: [ '/devdocs' ],
-		module: 'devdocs',
-		secondary: true,
-		enableLoggedOut: true
-	} );
+sections.push( {
+	name: 'devdocs',
+	paths: [ '/devdocs' ],
+	module: 'devdocs',
+	secondary: true,
+	enableLoggedOut: true
+} );
+
+sections.push( {
+	name: 'devdocs',
+	paths: [ '/devdocs/start' ],
+	module: 'devdocs',
+	secondary: false,
+	enableLoggedOut: true
+} );
 
-	sections.push( {
-		name: 'devdocs',
-		paths: [ '/devdocs/start' ],
-		module: 'devdocs',
-		secondary: false,
-		enableLoggedOut: true
-	} );
-}
 module.exports = sections.concat( extensionSections.filter( Boolean ) );
```

Comment:
is there a need to directly mutate `sections` anyway? could we get by with an immutable update that doesn't muck around with entirely separate modules?

```js
module.exports = sections
	.concat( [ {
		name: 'devdocs',
		paths: [ '/devdocs' ],
		…
	}, {
		name: 'devdocs',
		paths: [ '/devdocs/start' ],
		…
	} ] )
	.concat( extensionSections.filter( Boolean ) );
```